autoload/cmake4vim.vim

```diff
@@ -177,6 +176,7 @@ endfunction
 " Functions allows to switch between build types
 function! cmake4vim#SelectBuildType(buildType) abort
     let g:cmake_build_type = a:buildType
+    let g:cmake_usr_args   = utils#cmake#getCMakeVariants()[ a:buildType ][ 'cmake_usr_args' ]
```

Comment:
For future improvement: not only `cmake_usr_args` can be different.
I think different variants can also change all CMake build specific options:

- `cmake_install_prefix`
- `cmake_c_compiler`
- `cmake_cxx_compiler`

Reply:
That's actually a part of my next PR, it's introduced through `cmake-kits`.

Path to toolchain, specific compilers, environment variables, preferred generator.

As for the `cmake_install_prefix`, I'll put that in with the next PR too, I have something in the form of this:
```
g:cmake_build_path_pattern - pattern for build dir, 
                             two strings that will be evaluated in a printf and used 
                             instead of g:cmake_build_dir_prefix
e.g.: 
let g:cmake_build_path_pattern = 
     \ [ "%s/workspace/build/%s/%s/%s", 
     \   "$HOME, fnamemodify( getcwd(), ':t' ), g:cmake_selected_kit, g:cmake_build_type" ]
```